File: mcp_server_development/node/src/session03/answers/q2-server.ts
/**
 * 問題2 の解答：3 つの欠陥を修正した stdio エントリーポイント
 */
import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";

// 欠陥①の修正: ESM では相対 import に .js 拡張子が必要
import { createDashboardServer } from "../create-server.js";

const server = createDashboardServer();
const transport = new StdioServerTransport();

// 欠陥③の修正: トランスポートに接続していなかった
await server.connect(transport);

// 欠陥②の修正: サーバープロセスの stdout は JSON-RPC の通信路。ログは stderr へ
console.error("[q2-server] stdio でリクエストを待機しています");

File: mcp_server_development/python/src/session04/q2_server.py
"""問題2: 修正版の stdio エントリーポイント

修正点は 3 つです（詳細は解説の表）。
  1. import を実行時に解決できる形に直した
  2. ログの出力先を stdout から stderr に変えた
  3. mcp.run(transport="stdio") を呼んで待機状態に入るようにした
"""

import sys

from create_server import mcp

if __name__ == "__main__":
    print("[q2-server] stdio でリクエストを待機しています", file=sys.stderr)
    mcp.run(transport="stdio")
